### PR TITLE
Fix: Added missing call to _reject in Transactor#transaction

### PR DESCRIPTION
**Pull request by briandamaged**

### CODE REVIEW

File: lib/transaction.js [hunk 1]
```diff
@@ -280,13 +280,7 @@ function makeTransactor(trx, connection, trxClient) {
...
     } else {
       return new Promise((resolve, _reject) => {
-        trxClient.transaction(
-          (nestedTrx) => {
-            resolve(nestedTrx);
-          },
-          options,
-          trx
-        );
+        trxClient.transaction(resolve, options, trx).catch(_reject);
```
Comment (kibertoad) [inline lib/transaction.js]: in which case was this triggered? is it difficult to simulate in a test?

Comment (briandamaged) [reply lib/transaction.js]: Hmm... I can probably come up w/ a test case for it.  I'll give it a try in a few minutes.

Comment (briandamaged) [reply lib/transaction.js]: Thanks for reviewing all of these, btw!  Any hesitations before we merge this change?

**Comment by kibertoad** [comment]
I regret nothing.

**Pull request merged**